src/Raven.Server/Rachis/Remote/RemoteConnection.cs

```diff
@@ -270,66 +269,40 @@ public void Send(JsonOperationContext context, Exception e)
             });
         }
 
-        public unsafe int Read(byte[] buffer, int offset, int count)
+        internal int Read(byte[] buffer, int offset, int count)
         {
             using (_disposerLock.EnsureNotDisposed())
             {
-                if (_buffer.Used < _buffer.Valid)
-                {
-                    var size = Math.Min(count, _buffer.Valid - _buffer.Used);
-                    fixed (byte* pBuffer = buffer)
-                    {
-                        Memory.Copy(pBuffer + offset, _buffer.Pointer + _buffer.Used, size);
-                        _buffer.Used += size;
-                        return size;
-                    }
-                }
+                var read = ReadFromBuffer(buffer, offset, count);
+                if (read != -1) 
```

Comment:
Why is this change? It wasn't before, right?

Reply:
I just refactored the read from buffer code to a dedicated function

Reply:
okay, now I understand... I think a cleaner way will be something like this:
```
if (_buffer.Used < _buffer.Valid)
    return ReadFromBuffer(buffer, offset, count);

return _stream.Read(buffer, offset, count);
```